### ISSUE TITLE
More flexible "move" command in thread mode

### ISSUE THREAD

**Issue by pacien** [pazz/alot#1490]
**Is your feature request related to a problem? Please describe.**

Currently, the "move" command in thread mode only accepts notmuch queries with the "next" and "previous" subcommand.

It doesn't support:
* combining multiple subcommands to allow something like: "move first unfolded"
* the use notmuch queries with other subcommands, like: "move first tag:unread"

This is quite inconvenient and prevents the user from defining shortcuts such as "select; move first tag:unread" in the search mode to open a thread and focus directly on the first unread message [1]. 

**Describe the solution you'd like**

The "move" command could be made more flexible to allow the combination of multiple filters (currently subcommands) and the use of notmuch queries.

**Describe alternatives you've considered**

In the scenario [1]: using "select; move next tag:unread" moves to the second unread message in new threads, which is not the desired effect.

**Additional context**

Turning subcommands into filters could actually simplify the implementation of that command.

**Comment by pazz** [comment]
Yes, that'd be cool. Also, It be gread to write this somehow so that it doesn't blow up the code to 10000 lines. Suggestions welcome :)
